src/createServer.js

```diff
@@ -1,3 +1,63 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase'); // Импорт бизнес-логики
+
+function createServer() {
+  return http.createServer((req, resp) => {
+    resp.setHeader('Content-Type', 'application/json');
+
+    const parsedUrl = new URL(req.url, `http://${req.headers.host}`);
+    const queryParams = parsedUrl.searchParams;
+
+    const textToConvert = parsedUrl.pathname.slice(1);
+    const targetCase = queryParams.get('toCase');
+
+    const errors = [];
+
+    if (!textToConvert) {
+      errors.push({
+        message:
+          'Text to convert is required. Correct request is:' +
+          ' "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!targetCase) {
+      errors.push({
+        message:
+          '"toCase" query param is required. Correct request is:' +
+          ' "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    const validCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+    if (targetCase && !validCases.includes(targetCase)) {
+      errors.push({
+        message: `This case is not supported. Available cases: ${validCases.join(', ')}.`,
+      });
+    }
+
+    if (errors.length > 0) {
+      resp.end(JSON.stringify({ errors }));
```

Comment:
When there are errors, the response should include a 400 status code and 'Bad Request' status text, as specified in the task requirements .